Client.js

```diff
@@ -0,0 +1,58 @@
+const axios = require('axios');
+const userInRollout = require('./userIdHash');
+
+const WAYPOST_API_ADDRESS = "http://localhost:5000/";
+
+class Client {
+  constructor(config, context=undefined) {
+    this.config = config;
+    this.context = context;
+    this.featureFlags = {};
+  }
+
+  // Developer must specify a userId key in their argument object
+  addContext(contextObj) {
+    if (!contextObj || !contextObj.userId) {
+      throw new TypeError("Function must take an object containing a userId property");
+    }
+    this.context = contextObj;
+  }
+
+  async startPolling() {
+    while (true) {
+      await this.poll();
+      await this.delay();
+    }
+  }
+
+  async delay() {
+    await new Promise(resolve => setTimeout(resolve, this.config.pollingInterval));
+  }
+
+  async poll() {
+    const res = await axios.get(`${WAYPOST_API_ADDRESS}api/flags?sdk_key=${this.config.sdkKey}`);
+    res.data.forEach(flag => {
+      this.featureFlags[flag.name] = flag;
+    });
+  }
+
+  // defaultVal is a boolean
+  getFeature(featureName, defaultVal) {
```

Comment:
I'm wondering if `getFeature` is the best name for this. It's hard to glean the method's meaning right away. Maybe `evaluateFlag` or something similar?